baseplate/experiments/__init__.py

```diff
@@ -189,8 +201,36 @@ def variant(self, name, user=None, bucketing_event_override=None,
 
         return variant
 
+    def expose(self, experiment_name, variant_name, user=None, **kwargs):
+
+        experiment = self._get_experiment(experiment_name)
+
+        if experiment is None:
+            return
+
+        inputs = dict(kwargs)
+
+        if user:
+            if user.is_logged_in:
+                inputs["user_id"] = user.id
+                inputs["user_roles"] = user.roles
+            else:
+                inputs["user_id"] = user.loid
+            inputs["logged_in"] = user.is_logged_in
+            inputs['cookie_created_timestamp'] = user.event_fields().get('cookie_created')
```

Comment:
`user.event_fields` should probably already do all this work for you, can you put this work in there and just use `event_fields`?

Reply:
done.